tests/networkpolicy_test.go

```diff
@@ -1,173 +1,264 @@
+				By("Create deny-by-label networkpolicy")
+				policy = &networkv1.NetworkPolicy{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serverVMI.Namespace,
+						Name:      "deny-by-label",
+					},
+					Spec: networkv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"type": "test",
+							},
 						},
+						Ingress: []networkv1.NetworkPolicyIngressRule{},
 					},
-					Ingress: []v1network.NetworkPolicyIngressRule{},
-				},
-			}
-			var err error
-			policy, err = virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
-			Expect(err).ToNot(HaveOccurred())
-		})
+				}
+				createNetworkPolicy(policy)
+			})
 
-		AfterEach(func() {
-			waitForNetworkPolicyDeletion(policy)
-		})
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
 
-		It("[test_id:1515] should be failed to reach vmia from vmic", func() {
-			By("Connect vmia from vmic")
-			assertPingFail(vmic, vmia)
-		})
+			It("[test_id:1515] should be failed to reach serverVMI from clientVMIAlternativeNamespace", func() {
+				By("Connect serverVMI from clientVMIAlternativeNamespace")
+				assertPingFail(clientVMIAlternativeNamespace, serverVMI)
+			})
 
-		It("[test_id:1516] should be failed to reach vmia from vmib", func() {
-			By("Connect vmia from vmib")
-			assertPingFail(vmib, vmia)
-		})
+			It("[test_id:1516] should be failed to reach serverVMI from clientVMI", func() {
+				By("Connect serverVMI from clientVMI")
+				assertPingFail(clientVMI, serverVMI)
+			})
 
-		It("[test_id:1517] should be successful to reach vmib from vmic", func() {
-			By("Connect vmib from vmic")
-			assertPingSucceed(vmic, vmib)
+			It("[test_id:1517] should be successful to reach clientVMI from clientVMIAlternativeNamespace", func() {
+				By("Connect clientVMI from clientVMIAlternativeNamespace")
+				assertPingSucceed(clientVMIAlternativeNamespace, clientVMI)
+			})
+
+		})
+		Context("and an allow all http ports policy is populated", func() {
+			var policy *networkv1.NetworkPolicy
+			BeforeEach(func() {
+				port80 := intstr.FromInt(80)
+				port81 := intstr.FromInt(81)
+				tcp := corev1.ProtocolTCP
+				policy = &networkv1.NetworkPolicy{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serverVMI.Namespace,
+						Name:      "allow-all-http-ports",
+					},
+					Spec: networkv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{},
+						Ingress: []networkv1.NetworkPolicyIngressRule{
+							{
+								Ports: []networkv1.NetworkPolicyPort{
+									{Port: &port80, Protocol: &tcp},
+									{Port: &port81, Protocol: &tcp},
+								},
+							},
+						},
+					},
+				}
+				createNetworkPolicy(policy)
+			})
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
+			It("[test_id:2774] should allow http traffic for ports 80 and 81 from clientVMI to serverVMI", func() {
```

Comment:
i think we should add some tests that shows other ports are not reachable.

do we want to also add some rule that will allow it just from one namespace and test it ? (maybe this is too much)

Reply:
This PR is about having at kubevirt the same tier2 test we do at cnv-tests and that has a polarion test with this specification, is not about adding new tests.

Reply:
but see "do we want to also add some rule that will allow it just from one namespace and test it ? (maybe this is too much)" as also Miguel said on other thread, and i said above

https://github.com/kubevirt/kubevirt/pull/4061/files#r488062769

Reply:
AFAIU, the test @oshoval mentions is https://github.com/kubevirt/kubevirt/pull/4061/files#diff-bdaa78c5363f61af93fe1376b26fd2a4R255 (open one port, check port is accessible, while another port isn't).

Would adding yet another port achieve something I'm not seeing ? I could be missing something.

Reply:
There were 2 comments
1. combine the two tests , the one that allow 80+81 , and the one that allow 80 and checks 81 fails, into one,+ add 82 as the one that should fail - instead of having two tests. - as Quique explained, this PR align with polarion so no need.
Just took as invariant that we want 2 open ports, so added a 3rd to check it doesnt ping to it - anyway no need.

2. checking on different namespace, as Miguel also said on other thread.

Reply:
Added different namespaces.

Reply:
After thinking this a bit more (and after our discussion), I am unsure you need / should actually further complicate these tests out; at the end of the day, this is a k8s feature. AFAIU, we do nothing in KubeVirt to support that.

You can add those for completeness / get a better understanding on how it works, but I am no longer pushing for those.

Reply:
Well test is really slow already let's reduce the cadence of it remove this redundancies.